File: src/main/java/me/miki/shindo/management/mods/settings/impl/combo/Option.java
package me.miki.shindo.management.mods.settings.impl.combo;

import me.miki.shindo.management.language.TranslateText;

public class Option {

    private final TranslateText nameTranslate;

    public Option(TranslateText nameTranslate) {
        this.nameTranslate = nameTranslate;
    }

    public String getName() {
        return nameTranslate.getText();
    }

    public String getNameKey() {
        return nameTranslate.getKey();
    }

    public TranslateText getTranslate() {
        return nameTranslate;
    }
}

File: src/main/kotlin/com/shindoclient/shindo/management/settings/impl/combo/Option.kt
package com.shindoclient.shindo.management.settings.impl.combo

import com.shindoclient.shindo.management.language.TranslateText
import com.shindoclient.shindo.management.settings.Setting

class Option {
    private val nameTranslate: TranslateText?
    private val fallbackName: String
    val nameKey: String

    constructor(nameTranslate: TranslateText) {
        this.nameTranslate = nameTranslate
        this.fallbackName = nameTranslate.getText()
        this.nameKey = nameTranslate.getKey()
    }

    constructor(name: String) {
        this.nameTranslate = null
        this.fallbackName = name
        this.nameKey = buildKey(name)
    }

    val name: String
        get() = nameTranslate?.getText() ?: fallbackName

    fun getTranslate(): TranslateText? = nameTranslate

    private fun buildKey(raw: String?): String = if (raw == null) "" else Setting.normalizeKey(raw)
}
